aws-distro-opentelemetry-node-autoinstrumentation/src/aws-metric-attribute-generator.ts

```diff
@@ -411,9 +450,18 @@ export class AwsMetricAttributeGenerator implements MetricAttributeGenerator {
       remoteResourceIdentifier = AwsMetricAttributeGenerator.getDbConnection(span);
     }
 
-    if (remoteResourceType !== undefined && remoteResourceIdentifier !== undefined) {
+    if (cloudFormationIdentifier === undefined) {
+      cloudFormationIdentifier = remoteResourceIdentifier;
```

Comment:
Same as above^

Reply:
Let's get clarification from team for this specification.

Reply:
Sounds good!

Reply:
[Related](https://github.com/aws-observability/aws-otel-js-instrumentation/pull/121#discussion_r1866283099)

Agree, it probably doesn't make the most sense to set the CFN Id to connection string here. This condition was originally meant to default the CFN Id to the existing remote resource Id, as in most cases, they are the same. 

Unless this will cause some bug, I think the tradeoff here is one conditional for an extra unused attribute in DDB connection string vs. no conditional and manually setting the CFN for each case. 

Personally, I think the former is much more maintainable especially once we start supporting more AWS resources in the future. What do others think?

Reply:
We can avoid manually setting the CFN for each case by catching all `AWS resource` cases with a `isAwsSDKSpan(span)` check (already has usage in this method). I think better be safe than sorry here.

See https://github.com/aws-observability/aws-otel-js-instrumentation/pull/121#discussion_r1866370956

Reply:
Agree, I like this approach. Thanks for the suggestion/improvement!